File: ft/lua/black-atom-medical.lua
local M = {}

---@class BlackAtom.Theme.Meta
M.meta = {
    key = "black-atom-medical",
    label = "Black Atom - Medical",
    appearance = "light",
    status = "release",
    collection = {
        key = "black_atom",
        label = "Black Atom",
    },
}

---@class BlackAtom.Theme.Primaries
M.primaries = {
    "#222325",
    "#313533",
    "#3e4542",
    "#4a514f",
    "#545d5a",
    "#616a65",
    "#6d7873",
    "#7a8782",
    "#bcc3c0",
    "#c8cfcd",
    "#d4dbd8",
    "#e6e8e6",
}

---@class BlackAtom.Theme.Palette
M.palette = {
    black = "#313533",
    gray = "#545d5a",
    dark_red = "#5d54ad",
    red = "#7872c2",
    dark_green = "#628942",
    green = "#6e9a4b",
    dark_yellow = "#4a514f",
    yellow = "#57ab52",
    dark_blue = "#616a65",
    blue = "#545d5a",
    dark_magenta = "#887bd3",
    magenta = "#998ed9",
    dark_cyan = "#459f55",
    cyan = "#4db15e",
    light_gray = "#7a8782",
    white = "#d4dbd8",
}

---@class BlackAtom.Theme.Semantics
M.semantics = {
    bg = {
        primary = {
            dark = "#222325",
            main = "#d4dbd8",
            light = "#e6e8e6",
        },
        secondary = {
            dark = "#6d7873",
            main = "#7a8782",
            light = "#bcc3c0",
        },
        active = "#545d5a",
        match = {
            active = "#c8cfcd",
            passive = "#d4dbd8",
        },
        diff = {
            add = "#6e9a4b",
            delete = "#7872c2",
            change = "#545d5a",
            text = "#616a65",
        },
    },
    fg = {
        primary = {
            dark = "#222325",
            main = "#313533",
            light = "#3e4542",
        },
        secondary = {
            dark = "#4a514f",
            main = "#545d5a",
            light = "#616a65",
        },
        active = "#4db15e",
        invert = "#d4dbd8",
        neutral = "#6d7873",
        diff = {
            add = "#6e9a4b",
            delete = "#7872c2",
            change = "#545d5a",
            text = "#616a65",
        },
    },
}

---@class BlackAtom.Theme.Syntax
M.syntax = {
    default = "#313533",
    variable = {
        default = "#545d5a",
        builtin = "#616a65",
        parameter = "#4a514f",
        member = "#545d5a"
    },
    constant = {
        default = "#887bd3",
        builtin = "#998ed9",
    },
    module = {
        default = "#616a65",
    },
    string = {
        default = "#6e9a4b",
        documentation = "#628942",
        regexp = "#57ab52",
        escape = "#7872c2"
    },
    boolean = {
        default = "#998ed9"
    },
    number = {
        default = "#998ed9"
    },
    type = {
        default = "#4db15e",
        builtin = "#459f55"
    },
    attribute = {
        default = "#4db15e",
        builtin = "#459f55"
    },
    func = {
        default = "#57ab52",
        builtin = "#4a514f",
        method = "#57ab52"
    },
    constructor = {
        default = "#57ab52"
    },
    operator = {
        default = "#7872c2"
    },
    keyword = {
        default = "#7872c2"
    },
    punctuation = {
        default = "#545d5a",
        delimiter = "#545d5a",
        bracket = "#545d5a",
        special = "#4db15e"
    },
    comment = {
        default = "#4a514f",
        documentation = "#545d5a"
    },
    markup = {
        default = "#313533",
        heading = {
            default = "#616a65",
            h1 = "#616a65",
            h2 = "#616a65",
            h3 = "#545d5a",
            h4 = "#545d5a",
            h5 = "#545d5a",
            h6 = "#545d5a",
        },
        strong = "#222325",
        italic = "#313533",
        strikethrough = "#545d5a",
        underline = "#313533",
        quote = "#545d5a",
        math = "#459f55",
        link = {
            default = "#545d5a",
            label = "#6e9a4b",
            url = "#616a65"
        },
        code = {
            default = "#6e9a4b",
            inline = "#628942",
            block = "#6e9a4b",
        },
        list = {
            default = "#313533",
            checked =  "#4db15e",
            listUnchecked = ""
        },
    },
    diff = {
        plus = "#6e9a4b",
        minus = "#7872c2",
        delta = "#545d5a",
        text = ""
    },
    tag = {
        default = "#545d5a",
        builtin = "#616a65",
        custom = "#4db15e",
        attribute = "#6e9a4b",
        delimiter = "#7872c2"
    }
}

---@class BlackAtom.Theme.Colors
M.colors = {
    none = "none",
    debug = M.debug,
    primaries = M.primaries,
    palette = M.palette,
    semantics = M.semantics,
    syntax = M.syntax,
}

return M

File: src/themes/black-atom-medical.ts
import * as Theme from "../types/theme";

const meta: Theme.Meta = {
  key: "black-atom-medical",
  label: "Black Atom - Medical",
  appearance: "light",
  status: "release",
  collection: {
    key: "default",
    label: "Default"
  }
};

const primaries: Theme.Primaries = [
  // Lows
  "#222325",
  "#313533",
  "#3e4542",
  "#4a514f",

  // Mids
  "#545d5a",
  "#616a65",
  "#6d7873",
  "#7a8782",

  // Highs
  "#bcc3c0",
  "#c8cfcd",
  "#d4dbd8",
  "#e6e8e6"
];

const palette: Theme.Palette = {
  black: primaries[1],
  gray: primaries[4],
  darkRed: "#5d54ad",
  red: "#7872c2",
  darkGreen: "#628942",
  green: "#6e9a4b",
  darkYellow: primaries[3],
  yellow: "#57ab52",
  darkBlue: primaries[5],
  blue: primaries[4],
  darkMagenta: "#887bd3",
  magenta: "#998ed9",
  darkCyan: "#459f55",
  cyan: "#4db15e",
  lightGray: primaries[7],
  white: primaries[10]
};

const semantics: Theme.Semantics = {
  bg: {
    primary: {
      dark: primaries[9],
      main: primaries[10],
      light: primaries[11]
    },
    secondary: {
      dark: primaries[6],
      main: primaries[7],
      light: primaries[8]
    },
    active: primaries[4],
    match: {
      active: primaries[9],
      passive: primaries[10]
    },
    diff: {
      add: palette.green,
      delete: palette.red,
      change: palette.blue,
      text: palette.darkBlue
    }
  },
  fg: {
    primary: {
      dark: primaries[0],
      main: primaries[1],
      light: primaries[2]
    },
    secondary: {
      dark: primaries[3],
      main: primaries[4],
      light: primaries[5]
    },
    active: palette.cyan,
    invert: primaries[10],
    neutral: primaries[6],
    diff: {
      add: palette.green,
      delete: palette.red,
      change: palette.blue,
      text: palette.darkBlue
    }
  }
};

const syntax: Theme.Syntax = {
  default: semantics.fg.primary.main,
  variable: {
    default: palette.blue,
    builtin: palette.darkBlue,
    parameter: palette.darkYellow,
    member: palette.blue
  },
  constant: {
    default: palette.darkMagenta,
    builtin: palette.magenta
  },
  module: {
    default: palette.darkBlue
  },
  string: {
    default: palette.green,
    doc: palette.darkGreen,
    regexp: palette.yellow,
    escape: palette.red
  },
  boolean: {
    default: palette.magenta
  },
  number: {
    default: palette.magenta
  },
  type: {
    default: palette.cyan,
    builtin: palette.darkCyan
  },
  attribute: {
    default: palette.cyan,
    builtin: palette.darkCyan
  },
  func: {
    default: palette.yellow,
    builtin: palette.darkYellow,
    method: palette.yellow
  },
  constructor: {
    default: palette.yellow
  },
  operator: {
    default: palette.red
  },
  keyword: {
    default: palette.red
  },
  punctuation: {
    default: semantics.fg.secondary.main,
    delimiter: semantics.fg.secondary.main,
    bracket: semantics.fg.secondary.main,
    special: palette.cyan
  },
  comment: {
    default: semantics.fg.secondary.dark,
    documentation: semantics.fg.secondary.main
  },
  markup: {
    default: semantics.fg.primary.main,
    heading: {
      default: palette.darkBlue,
      h1: palette.darkBlue,
      h2: palette.darkBlue,
      h3: palette.blue,
      h4: palette.blue,
      h5: palette.blue,
      h6: palette.blue
    },
    strong: semantics.fg.primary.dark,
    italic: semantics.fg.primary.main,
    strikethrough: semantics.fg.secondary.main,
    underline: semantics.fg.primary.main,
    quote: semantics.fg.secondary.main,
    math: palette.darkCyan,
    link: {
      default: palette.blue,
      label: palette.green,
      url: palette.darkBlue
    },
    code: {
      default: palette.green,
      inline: palette.darkGreen,
      block: palette.green
    },
    list: {
      default: semantics.fg.primary.main,
      checked: semantics.fg.active,
      unchecked: semantics.fg.neutral
    }
  },
  diff: {
    neutral: semantics.fg.diff.text,
    plus: semantics.bg.diff.add,
    minus: semantics.bg.diff.delete,
    delta: semantics.bg.diff.change
  },
  tag: {
    default: palette.blue,
    builtin: palette.darkBlue,
    custom: palette.cyan,
    attribute: palette.green,
    delimiter: palette.red
  }
};

const theme: Theme.Definition = {
  meta,
  primaries,
  palette,
  semantics,
  syntax
};

export default theme;
